ethcore/light/src/net/mod.rs

```diff
@@ -1089,20 +1134,24 @@ impl NetworkProtocolHandler for LightProtocol {
 fn encode_request(req: &Request, req_id: usize) -> Vec<u8> {
 	match *req {
 		Request::Headers(ref headers) => {
-			let mut stream = RlpStream::new_list(5);
+			let mut stream = RlpStream::new_list(2);
+			stream.append(&req_id).begin_list(4);
+				
+			match headers.start {
+				HashOrNumber::Hash(ref hash) => stream.append(hash),
+				HashOrNumber::Number(ref num) => stream.append(num),
+			};
+			
 			stream
-				.append(&req_id)
-				.begin_list(2)
-					.append(&headers.block_num)
-					.append(&headers.block_hash)
 				.append(&headers.max)
 				.append(&headers.skip)
 				.append(&headers.reverse);
+
 			stream.out()
 		}
 		Request::Bodies(ref request) => {
-			let mut stream = RlpStream::new_list(request.block_hashes.len() + 1);
-			stream.append(&req_id);
+			let mut stream = RlpStream::new_list(2);
+			stream.append(&req_id).begin_list(request.block_hashes.len());
```

Comment:
Spec has been changed so that all requests and responses will have a variable-length list as the last item rather than having the root list be variable-length. 

(personally not the way that I would go about it but geth's behavior was not per the spec and I wanted to test the server with its light client)